pkg/modbus/device/twindata.go

```diff
@@ -63,5 +73,106 @@ func (td *TwinData) Run() {
 		klog.Error(err)
 	}
 
-	klog.V(2).Infof("Update value: %s, topic: %s", strconv.Itoa(int(td.Results[0])), td.Topic)
+	klog.V(2).Infof("Update value: %s, topic: %s", sData, td.Topic)
+}
+
+func TransferBytes(isRegisterSwap bool, isSwap bool, value []byte) []byte {
```

Comment:
ok,I will take a look.

Reply:
The following is my understanding, and my code can achieve it：
AB CD  : isRegisterSwap:false  isSwap:false
CD AB  : isRegisterSwap:true  isSwap:false
BA DC  : isRegisterSwap:false  isSwap:trur
DC BA  : isRegisterSwap:true  isSwap:true

But according to the code logic，when isRegister is true ，it will be DC BA！

I'm a little confused, can you explain these two parameters for me?

Reply:
Your implement is right.
However, it's more clear to split into two functions: swapRegister and swapByte.

Reply:
Ok, i will fix it!